src/hooks/useDetails.tsx

```diff
@@ -1,13 +1,23 @@
 import {useCallback, useEffect, useState, useRef} from 'react'
 
-function useDetails({ref, closeOnOutsideClick, defaultOpen, onClickOutside} = {}) {
+type useDetailsParameters = {
+  ref?: {current: HTMLElement}
+  closeOnOutsideClick?: boolean
+  defaultOpen?: boolean
+  onClickOutside?: Function
```

Comment:
I think we can be more specific here:

```suggestion
  onClickOutside?: (event: MouseEvent) => void
```